src/stcal/jump/jump.py

```diff
@@ -427,6 +427,13 @@ def detect_jumps(
             # save the neighbors to be flagged that will be in the next slice
             previous_row_above_gdq = row_above_gdq.copy()
             k += 1
+            # remove redundant bits in pixels that have jump flagged but were
+            # already flagged as do_not_use or saturated.
+            gdq[gdq == np.bitwise_or(dqflags['DO_NOT_USE'], dqflags['JUMP_DET'])] = \
+                dqflags['DO_NOT_USE']
+            gdq[gdq == np.bitwise_or(dqflags['SATURATED'], dqflags['JUMP_DET'])] = \
```

Comment:
Could this be simplified in checking only for `JUMP_DET` since if the `SATURATED` flag is already set, there's no need to check for it?

Reply:
The pixels have already had the JUMP_DET added when the slices are combined using the rows above and below the slice. This addition does not check for existing flags. It is cleaner to fix the redundant flagging that to check each pixel before adding the jump flag. The redundant flagging happens about 1 in a million pixels.